spec/paper_trail/request_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module PaperTrail
+  ::RSpec.describe(Request, versioning: true) do
+    describe ".enabled_for_model?" do
+      it "returns true" do
+        expect(PaperTrail.request.enabled_for_model?(Widget)).to eq(true)
+      end
+    end
+
+    describe ".disable_model" do
+      it "sets enabled_for_model? to false" do
+        expect(PaperTrail.request.enabled_for_model?(Widget)).to eq(true)
+        PaperTrail.request.disable_model(Widget)
+        expect(PaperTrail.request.enabled_for_model?(Widget)).to eq(false)
+      end
+
+      after do
+        PaperTrail.request.enable_model(Widget)
+      end
+    end
+
+    describe ".enable_model" do
+      it "sets enabled_for_model? to true" do
+        PaperTrail.request.disable_model(Widget)
+        expect(PaperTrail.request.enabled_for_model?(Widget)).to eq(false)
+        PaperTrail.request.enable_model(Widget)
+        expect(PaperTrail.request.enabled_for_model?(Widget)).to eq(true)
+      end
+
+      after do
+        PaperTrail.request.enable_model(Widget)
+      end
+    end
+
+    describe ".whodunnit" do
+      context "when set to a proc" do
+        it "evaluates the proc each time a version is made" do
+          call_count = 0
+          described_class.whodunnit = proc { call_count += 1 }
+          expect(described_class.whodunnit).to eq(1)
+          expect(described_class.whodunnit).to eq(2)
+        end
+      end
+    end
+
+    describe ".with" do
+      context "block given" do
```

Comment:
Can you add tests for the validator here through the public `with` test?  That is, test that whodunnit, enabled for work (maybe enabled for is a good new test case since it's a regex), transaction id, and an invalid option.

Also can you change the initial value to be non nil so we can test something more useful, e.g. instead of `expect(described_class.whodunnit).to be_nil`, we can have `expect(described_class.whodunnit).to 'previous_whodunnit`